docs(changelog): add 8.5.0 release notes (#5148)
canonical/snapcraft: docs/reference/changelog.rst

  For a complete list of commits, check out the `X.Y.Z`_ release on GitHub.

8.5.0 (2024-Nov-19)
-------------------

Core
====

> * Remove the Snapcraft ``Dockerfile`` in favour of the `snapcraft-rocks`_

Reviewer: ```suggestion
* Remove the Snapcraft Dockerfile in favour of the `snapcraft-rocks`_
```

It's a subtle distinction, but "Dockerfile" is a generic noun rather than a file name, like how we say "README".

  registry. For more information, see the ``docker/README.md`` file.

Bases
#####

core24
""""""

* Add a ``--platform <platform>`` argument to the ``clean`` to filter
  which build environments to clean (`canonical/craft-application#425`_).

Plugins
#######

Poetry
""""""

* Add a new plugin for projects using `Poetry <https://python-poetry.org/>`_.
  For more information, see :doc:`Poetry plugin
  reference</reference/plugins/poetry_plugin>`.

Extensions
##########

Env-injector
""""""""""""

* Add a new experimental extension called
  `env-injector <https://github.com/canonical/snappy-env>`_. It simplifies
  passing environment variables to snaps at runtime using ``snap config`` or a

>   ``env`` file. For more information, see `the env-injector

Reviewer: ```suggestion
  ``.env`` file. For more information, see `the env-injector
```

  extension <https://forum.snapcraft.io/t/the-env-injector-extension/41477>`_

Command line
============

* Order lifecycle commands in the sequence they run rather than alphabetically
  in help messages.

* Show positional arguments in help messages.

* Hide the terminal cursor when Snapcraft is running.

* Fix a bug where the version and path to the log file were logged multiple
  times.

* Fix a bug where deprecated commands were suggested when an invalid command
  was entered (`canonical/craft-cli#297`_).

Init
====

* Add a ``<project-dir>`` argument to initialise the project in a particular
  directory.

* Add a ``--name <name>`` argument to set the ``name`` key in the initialised

>   ``snapcraft.yaml``. If ``--name`` is not provided, the name of the project

Reviewer: ```suggestion
  ``snapcraft.yaml``. If ``--name`` isn't provided, the name of the project
```

  directory or current working directory is used.

* Add a ``--profile <profile>`` argument to initialise a project for a specific

>   use-case. Currently, only the ``simple`` profile is supported.

Reviewer: ```suggestion
  purpose. Currently, only the ``simple`` profile is supported.
```

"Use case" is on our [list of deprecated phrases](https://docs.ubuntu.com/styleguide/en/#words-and-phrases-to-avoid).

Store
=====

* Add support for ``registries`` with new commands ``list-registries`` and
  ``edit-registries``.

* Add support for specifying components in a validation set when using the
  ``edit-validation-sets`` command.

Documentation
=============

* Add :doc:`reference</reference/linters>` for linters.

> * Update the ``HACKING`` doc to instruct how to use ``tox``.

Reviewer: ```suggestion
* Update the ``HACKING`` doc with instructions for how to use ``tox``.
```

> * Update the changelog sidebar to hide subheadings underneath each release.

Reviewer: ```suggestion
* Hide subheadings in the changelog's table of contents.
```

* Add :doc:`reference</reference/plugins/poetry_plugin>` for
  the Poetry plugin.

> * Fix an issue were auto-generated command reference was not rendered

Reviewer: ```suggestion
* Fix an issue where the auto-generated command reference wasn't rendered
```
